pkg/controller/upload_controller.go

```diff
@@ -330,11 +331,27 @@ func patchUploadFailure(c *uploadController, req *pluginv1api.Upload, msg string
 	// update status to Failed
 	log := loggerForUpload(c.logger, req)
 	var err error
-	req, err = c.patchUpload(req, func(r *pluginv1api.Upload) {
-		r.Status.Phase = pluginv1api.UploadPhaseFailed
-		r.Status.CompletionTimestamp = &metav1.Time{Time: c.clock.Now()}
-		r.Status.Message = msg
-	})
+	if strings.Contains(msg, "Failed to clean up local snapshot") {
```

Comment:
It is not a good idead to use string match to identify error type. You can make two different methods for patching different failures.